code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingUtils.java

```diff
@@ -103,24 +104,26 @@ static List<Object> toList(final JSONArray jsonArray) throws JSONException {
     static String getChannelId(final Context context, final MessagingPushPayload payload) {
         final NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
 
-        if (payload == null || payload.getChannelId() == null || payload.getChannelId().isEmpty()) {
+        if (payload == null || StringUtils.isNullOrEmpty(payload.getChannelId())) {
             createDefaultNotificationChannel(context, notificationManager, payload);
-            return MessagingConstants.PushNotificationPayload.DEFAULTS.DEFAULT_CHANNEL_ID;
+            return MessagingConstants.PushNotificationPayload.DEFAULT.CHANNEL_ID;
         }
 
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
             return payload.getChannelId();
         }
 
         if (Build.VERSION.SDK_INT > Build.VERSION_CODES.O) {
-            if (notificationManager.getNotificationChannel(payload.getChannelId()) != null) {
-                return payload.getChannelId();
-            } else {
+            if (notificationManager.getNotificationChannel(payload.getChannelId()) == null) {
+                createCustomNotificationChannel(context, notificationManager, payload);
+            } else if (StringUtils.isNullOrEmpty(payload.getChannelId())) {
+                // only create a default channel if no channel id is specified in the push payload
                 createDefaultNotificationChannel(context, notificationManager, payload);
             }
+            return payload.getChannelId() != null ? payload.getChannelId() : MessagingConstants.PushNotificationPayload.DEFAULT.CHANNEL_ID;
         }
 
-        return MessagingConstants.PushNotificationPayload.DEFAULTS.DEFAULT_CHANNEL_ID;
+        return MessagingConstants.PushNotificationPayload.DEFAULT.CHANNEL_ID;
```

Comment:
is this line intended to be the behavior only for `Build.VERSION_CODES.O`?

Reply:
the returned channel id is only used if the android version is `O` or greater. 
```java
// Setting the channel id
        final String channelId = MessagingUtils.getChannelId(context, payload);
        final Notification.Builder notificationBuilder = new Notification.Builder(context);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            notificationBuilder.setChannelId(channelId);
        }```